pkg/ticker/ticker.go

```diff
@@ -88,12 +112,13 @@ func (t *Ticker) Run(ctx context.Context) (err error) {
 
 	// setup
 	t.ticker = time.NewTicker(t.interval)
-	t.signalChan = make(chan struct{})
+	t.internalStopChan = make(chan struct{})
```

Comment:
You could also wrap a the ctx `ctx, cancel := context.WithCancel(ctx)`, store the `cancel` function in the `Ticker` struct then call it on `Stop()`

Reply:
https://github.com/zeta-chain/node/pull/2896/commits/f5a4701dcbfec24408d727fc9cb64895ac25907f